packages/Search/examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -12,21 +12,126 @@
 #include <DTK_DistributedSearchTree.hpp>
 
 #include <Kokkos_DefaultNode.hpp>
-#include <Teuchos_CommHelpers.hpp>
 #include <Teuchos_CommandLineProcessor.hpp>
-#include <Teuchos_DefaultComm.hpp>
-#include <Teuchos_StandardCatchMacros.hpp>
-#include <Teuchos_Time.hpp>
-#include <Teuchos_TimeMonitor.hpp>
 
+#include <mpi.h>
+
+#include <algorithm>
+#include <chrono>
 #include <cmath> // cbrt
+#include <numeric>
 #include <random>
+#include <utility>
+#include <vector>
+
+// Poor man's replacement for Teuchos::TimeMonitor
+class TimeMonitor
+{
+    using container_type = std::vector<std::pair<std::string, double>>;
+    using entry_reference_type = container_type::reference;
+    container_type _data;
+
+  public:
+    class Timer
+    {
+        entry_reference_type _entry;
+        bool _started;
+        std::chrono::high_resolution_clock::time_point _tick;
+
+      public:
+        Timer( entry_reference_type ref )
+            : _entry{ref}
+            , _started{false}
+        {
+        }
+        void start()
+        {
+            assert( !_started );
+            _tick = std::chrono::high_resolution_clock::now();
+            _started = true;
+        }
+        void stop()
+        {
+            assert( _started );
+            std::chrono::duration<double> duration =
+                std::chrono::high_resolution_clock::now() - _tick;
+            // NOTE I have put much thought into whether we should use the
+            // operator+= and keep track of how many times the timer was
+            // restarted.  To be honest I have not even looked was the original
+            // TimeMonitor behavior is :)
+            _entry.second = duration.count();
+            _started = false;
+        }
+    };
+    std::unique_ptr<Timer> getNewTimer( std::string name )
+    {
+        _data.emplace_back( std::move( name ), 0. );
```

Comment:
Potential timer duplication with the current choice of container.

Reply:
I think that Teuchos logic returns the existing timer with this call. This allows one to accumulated timer time in a loop.

Reply:
I agree but won't fix.

Reply:
I added a comment to capture this